helix-core/src/main/java/org/apache/helix/manager/zk/ParticipantManager.java

```diff
@@ -242,13 +242,17 @@ private CloudInstanceInformation getCloudInstanceInformation() {
         _helixManagerProperty.getHelixCloudProperty().getCloudInfoProcessorName();
     try {
       // fetch cloud instance information for the instance
-      String cloudInstanceInformationProcessorClassName = CLOUD_PROCESSOR_PATH_PREFIX
-          + _helixManagerProperty.getHelixCloudProperty().getCloudProvider().toLowerCase() + "."
-          + cloudInstanceInformationProcessorName;
+      String cloudInstanceInformationProcessorClassName =
```

Comment:
I dont get this... Originally, user configured it as "AZURE" will natively mapping to Azure impl. Then if they are using "CUSTOMIZED" they should be able to use any class path. 

I understand what you are trying to do. But the branching of logic should be at the place selecting type instead of using path to do the selection.

Reply:
Yes, you are correct. The logic to construct the fully qualified path does not really belong in `ParticipantManager`. I have moved this to `HelixCloudProperty` instead. 

Based on the usage of `_cloudInfoProcessorName`, I also decided to deprecate that and create `_cloudInfoProcessorFullyQualifiedClassName`.

Now `ParticipantManager` will just call `getCloudInfoProcessorFullyQualifiedClassName` and attempt to load the class.